src/lib/common/gameInstance.ts

```diff
@@ -35,31 +57,52 @@ class GameInstance {
         { pos: [0, -75], dir: [0, 10] }
       ]
     });
+    this.world.add({
+      gameState: {
+        inLobby: true,
+        inGame: false,
+        showingLeaderboard: false,
+        controlsDisabled: true
+      }
+    });
+    this.gameStateManager = new GameStateManager(this.world);
   }
-
   tick(ctx: TickContext) {
     steeringSystem(this.world, this.localPlayerId);
     movementSystem(this.world, undefined, ctx.deltaTime);
     driftCorrectionSystem(this.world, ctx.deltaTime);
     roadCollisionSystem(this.world, ctx.deltaTime);
+    carCollisionSystem(this.world);
+    this.gameStateManager.tick();
   }
 
   addPlayer(playerId: string) {
+    if (this.availableCodenames.length === 0) {
+      console.log('No available codenames');
+      return null;
+    }
```

Comment:
Returning null here changes the function signature, can we handle that in `socketIOHandler.ts`? There seems to be an error the about the player being null, we could exit early if `addPlayer` returns null.